tests/python/test_matrix.py

```diff
@@ -27,6 +44,23 @@ def test_python_scope_matrix_operations():
         assert np.allclose(c.to_numpy(), ops(a, b))
 
 
+@ti.host_arch_only
+def test_taichi_scope_matrix_operations_with_global_matrices():
+    for ops in operation_types:
+        ti.reset()
+        a, b = test_matrix_arrays
+        m1, m2 = ti.Matrix(a), ti.Matrix(b)
+        c = ti.Matrix(2, 2, dt=ti.i32, shape=())
```

Comment:
How about:
```suggestion    
    c = ti.Matrix(2, 2, dt=ti.i32, shape=())
    for ops in operation_types:
        a, b = test_matrix_arrays
        m1, m2 = ti.Matrix(a), ti.Matrix(b)
```
To prevent `No new variables can be declared after kernel invocations or Python-scope tensor accesses`?